### PR TITLE
[ONNX] Fix bug in exporting node with multiple outputs by scripting

### PR DESCRIPTION
**Pull request by BowenBao**

### CODE REVIEW

File: test/onnx/test_pytorch_onnx_caffe2.py [hunk 1]
```diff
@@ -77,6 +77,11 @@ def wrapper(self):
 def do_export(model, inputs, *args, **kwargs):
     f = io.BytesIO()
     out = torch.onnx._export(model, inputs, f, *args, **kwargs)
+    if isinstance(model, torch.jit.ScriptModule):
```
Comment (houseroad) [inline test/onnx/test_pytorch_onnx_caffe2.py]: Can we fix `out` (such as unpack) instead of directly rerun the model?

**Review by houseroad — COMMENTED** [review]
Shall we also add example_outputs to the `export` function?

File: test/onnx/test_pytorch_onnx_caffe2.py (hunk 1 above)
Comment (BowenBao) [reply test/onnx/test_pytorch_onnx_caffe2.py]: I don't understand the "fix" in your comment... When exporting ScriptModule we don't run the model, we get `out=None`. The `out` here is used to compare against `caffe2_out`, so it shouldn't hurt running the model if it's ScriptModule?

**Comment by BowenBao** [comment]
> Shall we also add example_outputs to the `export` function?

added

File: test/onnx/test_pytorch_onnx_caffe2.py (hunk 1 above)
Comment (BowenBao) [reply test/onnx/test_pytorch_onnx_caffe2.py]: this part is not related to the "fix" in the PR title. It is basically updating the current test infra to be able to test ScriptModule models. The actual fix is in torch/csrc/jit/script/init.cpp _propagate_and_assign_input_and_output_shapes

**Review by houseroad — APPROVED** [review]
Looks good.

**Pull request closed without merging**